File: typescript/hamming/hamming.ts
export default class Hamming {

    compute(s1: string, s2: string): number {
        if (s1.length !== s2.length) {
            throw new Error('DNA strands must be of equal length.')
        }

        let hammingDistance: number = 0
        for (let i = 0; i < s1.length; i++) {
            if (s1[i] !== s2[i]) {
                hammingDistance += 1
            }
        }
        return hammingDistance
    }
}

File: csharp/hamming/Hamming.cs
using System;

public static class Hamming
{
    public static int Distance(string s1, string s2)
    {
        if (s1.Length != s2.Length) throw new ArgumentException();

        int hammingDistance = 0;

        for (int i = 0; i < s1.Length; i++)
        {
            if (s1[i] != s2[i]) hammingDistance++;
        }
        return hammingDistance;
    }
}
